requestParser/requestParser.go

```diff
@@ -332,3 +347,25 @@ func getBirthDate(payload []byte) (time.Time, bool) {
 	}
 	return birthDate, true
 }
+
+//use regex matching to see what type of event an event destination is, for events metrics
+func getEventDestinationType(eventDestination string) string {
+	if len(eventDestination) == 0 {
+		return noEventDestination
+	}
+
+	switch {
+	case regexp.MustCompile(onlineEventRegex).MatchString(eventDestination):
```

Comment:
A few suggestions for this function:
- [ ] `MustCompile()` is usually only used in unit tests, because it can panic.  Use [Compile()](https://golang.org/pkg/regexp/#Compile) instead.
- [ ] We can compile the regular expressions once and then always use the compiled regexp structs for matching, which will give better performance than compiling everything multiple times.  I would suggest doing that as a part of `NewRequestParser()` and then storing the compiled regular expressions as a `map[*regexp.Regexp]string` in the `RequestParser` struct.  Then this function could be something like:
```go
func getEventDestinationType(regexs map[*regexp.Regexp]string, val string) {
  // add empty val check
  for r, t := range regexs {
    if r.MatchString(val) {
      return t
    }
  }
  return otherEventDestination
}
```
- [ ] Rather than have all these constants and regular expressions hard coded, we might want to read this from the configuration file.  This is up to @kcajmagic.  If he wants to do this, I'm happy to explain more.